radix-engine/src/system/node_modules/access_rules/package.rs

```diff
@@ -587,68 +256,58 @@ impl AccessRulesNativePackage {
         Ok(Own(component_id))
     }
 
-    pub(crate) fn set_authority_access_rule<Y>(
-        object_key: ObjectKey,
-        authority_key: AuthorityKey,
-        rule: AccessRule,
+    fn update_role<Y>(
+        role_key: RoleKey,
+        rule: Option<AccessRule>,
+        mutability: Option<(RoleList, bool)>,
         api: &mut Y,
     ) -> Result<(), RuntimeError>
     where
         Y: ClientApi<RuntimeError>,
     {
-        Self::set_authority_access_rule_and_mutability_internal(
-            object_key,
-            authority_key,
-            Some(rule),
-            None,
-            api,
-        )
-    }
+        let handle = api.actor_lock_field(
```

Comment:
We don't need to check `SELF_ROLE` here?

Reply:
good catch